File: backend/Models/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
#nullable disable


namespace EcommerceApp.Models
{
    public class Product
    {
        public int Id { get; set; }

        [MaxLength(100)]
        public string Name { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal Price { get; set; }

        public uint Quantity { get; set; }

        [MaxLength(200)]
        public string ImageUrl { get; set; }

        public string VectorData { get; set; }
    }
}

File: frontend/src/types/product.ts
/*
Old product
export type Product = {
  title: string;
  slug: string;
  content: string;
  price: string;
  typeOfCake: TypeOfCake[];
  coverImage: string;
}
  */

export type Product = {
	name: string;
	slug: string;
	description: string;
	price: number;
	imageUrl: string;
	quantity: number;
	vectorData: string;
};
/*
export type TypeOfCake = {
  label: string;
  value:*/
